### DISCUSSION TITLE
How to use process.env in h3?

### DISCUSSION THREAD

**Discussion by cipami** [nuxt/nuxt#18216] [Questions] [answered]
what is the correct way to access `process.env` in server/api? when accessing directly `process.env.CUSTOM_ENV` it only works in dev but not after build. Are there any extra steps needed for build?

**Comment by wobsoriano** [comment, marked as answer]
In order to preserve config in build stage, the variables should be added in `privateRuntimeConfig` and imported from `#config`:

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
    privateRuntimeConfig: {
        apiKey: process.env.API_KEY
    }
})
```

```ts
// server/api/test.ts
import runtimeConfig from '#config'

export default (req, res) => {
  console.log(runtimeConfig.apiKey)
}
```

More info: https://github.com/nuxt/framework/discussions/1672#discussioncomment-1586585

**Reply by cipami** [reply]
It works perfectly. Thanks!
